### ISSUE TITLE
Podman REST API can't create container with mounts

### ISSUE THREAD

**Issue by msisj** [containers/podman#13717]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

I was trying to implement mounts handling in [podman-py project](https://github.com/containers/podman-py) but it is impossible to do it basing on REST API documentation.


**Steps to reproduce the issue:**

1. Check REST API documentation https://docs.podman.io/en/latest/_static/api.html#operation/ContainerCreateLibpod (mounts option)

2. Run `curl --unix-socket /run/podman/podman.sock --header "Content-Type: application/json" --request POST --data '{"Image":"localhost/ubi8:latest","Name": "mushroom", "mounts":[{ "Target":"/dev/mushroom","Type":"tmpfs","TmpfsOptions":{"SizeBytes":100,"Mode":0}}]}' http://localhost/v4.0.2/libpod/containers/create`

**Describe the results you received:**

{"cause":"container directory cannot be empty","message":"container directory cannot be empty","response":500}

**Describe the results you expected:**

Container with proper tmpfs.

**Additional information you deem important (e.g. issue happens only occasionally):**

Ok so here's full story.
It looks like the documentation for [3.2 REST API ](https://docs.podman.io/en/latest/_static/api.html?version=v3.2#operation/ContainerCreateLibpod) is valid for mount options (TmpfsOptions -> options, Target -> destination)
Proof : 

```
[root@localhost ~]# curl --unix-socket /run/podman/podman.sock --header "Content-Type: application/json" --request POST --data '{"Image":"localhost/ubi8:latest","Name": "snake","mounts":[{ "Destination":"/dev/snake","Type":"tmpfs","options":["size=100","mode=0"]}]}' http://localhost/v4.0.2/libpod/containers/create
{"Id":"522fc796090868e5121e194ed8b560e7179459dcf41a50dff5aa7128fc0b7fce","Warnings":[]}
[root@localhost ~]# podman inspect snake --format '{{json .HostConfig.Tmpfs}}'
{"/dev/snake":"size=100,mode=0,rw,rprivate,nosuid,nodev,tmpcopyup"}
```

I don't believe that's intentional.

So hopefully we can agree that REST API documentation is invalid ... or is it really ? 

```
[root@localhost ~]# curl --unix-socket /run/podman/podman.sock --header "Content-Type: application/json" --request POST --data '{"Image":"localhost/ubi8:latest","HostConfig":{"mounts":[{"Target":"/dev/badger","Type":"tmpfs","TmpfsOptions":{"SizeBytes":100,"Mode":0}}]}}' http://localhost/containers/create?name=badger
{"Id":"e9bda4cb9d47560c0d369f097829c9eccaaf9bf5452ea682987ac8bec58727cb","Warnings":[]}
[root@localhost ~]# podman inspect badger --format '{{json .HostConfig.Tmpfs}}'
{"/dev/badger":"size=100,mode=0,rw,rprivate,nosuid,nodev,tmpcopyup"}
```

It looks like the **other** endpoint actually can accept the 4.0 (or main) REST API documentation.
Please note the url difference : 
`http://localhost/v4.0.2/libpod/containers/create` vs `http://localhost/containers/create?name=badger`

I'm just a simple user of podman and I don't really understand what's the difference between these endpoints (and why there are 2 different endpoints (docker compatibility maybe?)).

Obviously same goes for any other mount type created that way.

**Output of `podman version`:**

```
[root@localhost ~]# podman version
Client:       Podman Engine
Version:      4.0.2
API Version:  4.0.2
Go Version:   go1.17.7

Built:      Wed Mar  2 20:22:48 2022
OS/Arch:    linux/amd64
```

**Output of `podman info --debug`:**

```
host:                                                                                                                                                                                                                           [56/1848]
  arch: amd64
  buildahVersion: 1.24.1
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - hugetlb
  - pids
  - rdma
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.32-1.module+el8.5.0+13852+150547f7.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.32, commit: 4b12bce835c3f8acc006a43620dd955a6a73bae0'
  cpus: 4
  distribution:
    distribution: '"rhel"'
    version: "8.5"
  eventLogger: journald
  hostname: localhost
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 4.18.0-348.20.1.el8_5.x86_64
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 340672512
  memTotal: 3897122816
  networkBackend: cni
  ociRuntime:
    name: runc
    package: runc-1.0.3-1.module+el8.5.0+13556+7f055e70.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.0.3
      spec: 1.0.2-dev
      go: go1.16.7
      libseccomp: 2.5.1
  os: linux
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: ""
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.8-1.module+el8.5.0+12582+56d94c81.x86_64
    version: |-
      slirp4netns version 1.1.8
      commit: d361001f495417b880f20329121e3aa431a8f90f
      libslirp: 4.4.0
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.1
  swapFree: 4293902336
  swapTotal: 4294963200
  uptime: 124h 43m 56.11s (Approximately 5.17 days)
plugins:
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 10
    paused: 0
    running: 6
    stopped: 4
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 8
  runRoot: /run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 4.0.2
  Built: 1646252568
  BuiltTime: Wed Mar  2 20:22:48 2022
  GitCommit: ""
  GoVersion: go1.17.7
  OsArch: linux/amd64
  Version: 4.0.2
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
[root@localhost ~]# rpm -q podman
podman-4.0.2-1.module+el8.6.0+14379+4ec2a99a.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**


I guess this is the latest version.

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by mheon** [comment]
The docs here are completely wrong - that is not the struct used. [This](https://github.com/opencontainers/runtime-spec/blob/main/specs-go/config.go#L110-L120) is what we're actually using.

@jwhonce I recall us seeing this before, is it fixed on newer versions?

**Comment by Luap99** [comment]
I think the problem is that the types used in the swagger docs are not namespaced when they are generated so it will use the same struct name only once.

**Comment by jwhonce** [comment]
@msisj A quick note on the URL formatting, 

- URLs like `vX.Y.Z/libpod/containers` are for the podman/libpod API endpoints. 
- Whereas URLs like `vX.Y.Z/containers` or `containers` are for compatibility endpoints.

`/_ping` is an out-layer, it always returns the compatibility and libpod Headers. This was done to allow clients to switch between endpoints if they desire.

**Comment by msisj** [comment]
@mheon Thanks! I guess that means that there is no plan to switch to the structure which is currently in REST API documentation.
@jwhonce I see, that's what I've suspected, it was pretty confusing for me that the other endpoint uses "new" REST API, but since it's a bug it uses just "different" syntax.

I think this issue should stay opened then up until the documentation (or most likely swagger, as pointed out by @Luap99) is fixed.
I've found another issue with mounts (REST API) but this time something is actually not working correctly so I think I'll open another issue.

**Comment by rhatdan** [comment]
@jwhonce @Luap99 What should we do with this issue?

**Comment by TTWShell** [comment]
What should we do with this issue? +1 @Luap99

**Comment by rhatdan** [comment]
@baude @mheon PTAL

**Comment by inknos** [comment]
posting an update to not leave this too quiet.

I tried to come up with a few solutions that basically were working but breaking changes.
The problem is simply to make the `swagger` understand that it should use our `spec.Mount` from the docs. Currently (and I **don't** know why) it seems impossible to redirect its attention to the right struct using `// swagger:model` or similar code comments
